pkg/upgrade/machine_creator.go

```diff
@@ -325,3 +302,8 @@ func WithNodeReadyTimeout(timeout time.Duration) MachineCreatorOption {
 		n.nodeReadyTimeout = timeout
 	}
 }
+func WithManagementClient(c ctrlclient.Client) MachineCreatorOption {
```

Comment:
What do you think about renaming this just to `Client` and the struct field to `client` as well?